Otus-DevOps-2018-02/ifqthenp_infra
# Homework 7. Terraform 1

# Completed Homework №7

:large_blue_diamond: [x] Main tasks
:large_orange_diamond: [x] Extra tasks for self-study

## The following tasks have been completed

- :large_blue_diamond: Created terraform configuration files `main.tf`, `outputs.tf`, `variables.tf`, and `terraform.tfvars` allowing reddit application to be deployed on GCP instance. Input variables have been defined in `terraform.tfvars` and `terraform.tfvars.example` file has been pushed to the repo
- :large_orange_diamond: Added terraform resource to manage GCP SSH keys on project level. The problem encountered with this approach is that terraform will rewrite all previously added keys and `terraform destroy` command will delete all keys from project metadata.

## How to get project running

- Create `terraform.tfvars` file in `terraform` directory and define your variables
- Run `terraform apply -auto-approve` from `terraform` directory

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Open in browser http://35.197.250.139:9292/

## PR checklisk
